### PR TITLE
Lesters/add model eval rest api

### PR DESCRIPTION
**Pull request by lesters**
@bratseth Please review.

### CODE REVIEW

File: model-evaluation/src/main/java/ai/vespa/models/handler/ModelsEvaluationHandler.java [hunk 1]
```diff
@@ -1,47 +1,209 @@
...
+    private HttpResponse listTypeDetails(HttpRequest request, String modelName, String signatureAndOutput) {
+        return listTypeDetails(request, modelsEvaluator.evaluatorOf(modelName, signatureAndOutput));
+    }
+
+    private HttpResponse listTypeDetails(HttpRequest request, FunctionEvaluator evaluator) {
+        Slime slime = new Slime();
+        Cursor root = slime.setObject();
+        Cursor bindings = root.setArray("bindings");
+        for (String bindingName : evaluator.context().names()) {
+            if (bindingName.startsWith("constant(")) {
```
Comment (bratseth) [inline model-evaluation/src/main/java/ai/vespa/models/handler/ModelsEvaluationHandler.java]: Ok, but I think we need another way to do this which only exposes the external bindings.

**Review by bratseth — APPROVED** [review]
LGTM!

**Comment by bratseth** [comment]
Also, I think most of the return status codes should be 404 not 400.

**Comment by bratseth** [comment]
You also need to add a package-info with @ExportPackage (but not @PublicApi) in the handler package.

**Review by bratseth — CHANGES_REQUESTED** [review]
be sure to add the package-info file before merging

**Review by bratseth — APPROVED** [review]

**Comment by bratseth** [comment]
I did my suggestions, merging.

**Comment by lesters** [comment]
Thanks! I'll refactor with the TODOs you've added in mind.

Regarding ExportPackage, I'm pretty sure this is not required for this handler as it is not supposed to be used programatically. I tested building, running and querying the API successfully before committing.

**Comment by bratseth** [comment]
How can dependency injection create it if it cannot access it?

**Comment by lesters** [comment]
Well, doesn't this also go for user created application packages? We don't require them the ExportPackage either?

**Comment by bratseth** [comment]
Good point; However, I think it is needed still because the handler is referenced in the config model.

**Comment by lesters** [comment]
That is true. Most probably this worked due to the fat jar in config model so it was contained there anyway. I will decouple this.

**Pull request merged**